pallets/score/src/lib.rs

```diff
@@ -0,0 +1,268 @@
+// This file is part of CORD – https://cord.network
+
+// Copyright (C) 2019-2022 Dhiway Networks Pvt. Ltd.
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+// CORD is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// CORD is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with CORD. If not, see <https://www.gnu.org/licenses/>.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+#![allow(clippy::unused_unit)]
+#![warn(unused_crate_dependencies)]
+
+use cord_primitives::{
+	ss58identifier, CountOf, IdentifierOf, ScoreIdentifierOf, ScoreOf, SCORE_PREFIX,
+};
+use frame_support::ensure;
+use sp_runtime::traits::{IdentifyAccount, Saturating, Verify};
+use sp_std::{prelude::Clone, str};
+
+pub mod scores;
+pub mod weights;
+
+pub use crate::scores::*;
+
+pub use crate::weights::WeightInfo;
+pub use pallet::*;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+	use frame_support::pallet_prelude::*;
+	use frame_system::pallet_prelude::*;
+
+	pub(crate) type CordAccountOf<T> = <T as frame_system::Config>::AccountId;
+	pub type EntitySignatureOf<T> = <T as Config>::Signature;
+	pub type BlockNumberOf<T> = <T as frame_system::Config>::BlockNumber;
+
+	/// business entity
+	pub type EntityIdentifierOf<T> = <T as frame_system::Config>::AccountId;
+	/// seller application
+	pub type RequestorIdentifierOf<T> = <T as frame_system::Config>::AccountId;
+	/// buyer application
+	pub type CollectorIdentifierOf<T> = <T as frame_system::Config>::AccountId;
+
+	pub type EntryHashOf<T> = <T as frame_system::Config>::Hash;
+	pub type JournalIdentifierOf = IdentifierOf;
+	pub type RequestIdentifierOf = ScoreIdentifierOf;
+	pub type TransactionIdentifierOf = ScoreIdentifierOf;
+
+	pub type JournalDetailsOf<T> = JournalDetails<
+		EntityIdentifierOf<T>,
+		RequestIdentifierOf,
+		TransactionIdentifierOf,
+		CollectorIdentifierOf<T>,
+		RequestorIdentifierOf<T>,
+		ScoreTypeOf,
+		ScoreOf,
+	>;
+	pub type JournalInputOf<T> =
+		JournalInput<JournalDetailsOf<T>, EntryHashOf<T>, EntitySignatureOf<T>>;
+	pub type JournalEntryOf<T> =
+		JournalEntry<JournalDetailsOf<T>, EntryHashOf<T>, BlockNumberOf<T>>;
+	pub type ScoreEntryOf = ScoreEntry<CountOf, ScoreOf>;
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {
```

Comment:
We have to change the account details to use DIDs

Reply:
Thanks. Still working on it.